### DISCUSSION TITLE
API authentication issues

### DISCUSSION THREAD

**Discussion by Mazawrath** [rawandahmad698/PyChatGPT#98] [Q&A]
Hi all,
I've been getting hardcore rate limited on my current IP address using PythonAnywhere, so I want to try out a proxy. When passing the proxy over to PyChatGPT I've been getting this error:
`tls_client.exceptions.TLSClientExeption: failed to do request: Get "https://chat.openai.com/auth/login": x509: certificate signed by unknown authority`.
What would be the way around this?

**Comment by Mazawrath** [comment]
I got past that by specifying the same proxy for both http and https, but now getting stuck at sending requests to OpenAI
`Error when calling OpenAI API: HTTPSConnectionPool(host='chat.openai.com', port=443): Max retries exceeded with url: /backend-api/conversation (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))`

I'm generally inexperienced with proxies and certificates so any help would be appreciated.

**Comment by rawandahmad698** [comment]
@Mazawrath Hi, this is because the certificate cannot verified using your proxy, normally you turn off verify=False, but currently Tls-Client does not support it directly. Although I have created a fork of it and I have lots of updates to push. In the meantime try a VPN?

**Reply by Mazawrath** [reply]
With this new Cloudflare protection I assume the login now needs to be on the same IP address as the `ask()` requests are made on.
I was thinking that there could be a feature where you login with your regular IP to get around the certificate issue, and then you can pass the certificate and proxy to `Requests` so that doesn't error out, but that's not doable now, is it?
